auth/auth-router.js

```diff
@@ -1,11 +1,59 @@
 const router = require('express').Router();
+const bcryptjs = require("bcryptjs");
+const jwt = require("jsonwebtoken");
+
+const configs = require("../api/configs");
+const Users = require("./auth-model");
+const { isValid } = require("./auth-services");
 
 router.post('/register', (req, res) => {
-  // implement registration
+  const credentials = req.body;
+  if(isValid(credentials)){
+    const rounds = process.env.BCRYPT_ROUNDS || 4;
+    const hash = bcryptjs.hashSync(credentials.password, rounds);
+    credentials.password = hash;
+    Users.add(credentials)
```

Comment:
I wouldn't mutate the user data this way, but rather pass it directly into `.add`. I also wouldn't blanket pass the whole body into `Users.add`, as a malicious user could add some property to the object that messes things up. Instead, I would make it explicit: `Users.add({ username: credentials.username, password: hash })`